frontend/public/components/deploy-image.tsx

```diff
@@ -0,0 +1,439 @@
+    this.onKeyPress = this.onKeyPress.bind(this);
+    this.onNameChange = this.onNameChange.bind(this);
+    this.search = this.search.bind(this);
+    this.save = this.save.bind(this);
+  }
+
+  onNamespaceChange = (namespace: string) => {
+    this.setState({namespace: namespace});
+  };
+
+  onImageNameChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({imageName: event.currentTarget.value});
+  }
+
+  onKeyPress = event => {
+    if (this.state.imageName !== '' && event.key === 'Enter') {
+      this.search(event);
+    }
+  }
+
+  onNameChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({name: event.currentTarget.value});
+  };
+
+  search = event => {
+    event.preventDefault();
+
+    const { namespace, imageName } = this.state;
+    const importImage = {
+      kind: 'ImageStreamImport',
+      apiVersion: 'image.openshift.io/v1',
+      metadata: {
+        name: imageName,
+        namespace: namespace
+      },
+      spec: {
+        import: false,
+        images: [{
+          from: {
+            kind: 'DockerImage',
+            name: imageName
+          }
+        }]
+      },
+      status: {}
+    };
+
+    this.setState({
+      loading: true,
+      isi: null,
+      name: '',
+      error: null,
+      searchError: null,
+    });
+
+    k8sCreate(ImageStreamImportsModel, importImage)
+      .then((imageStreamImport) => {
+        const name = _.get(imageStreamImport, 'spec.images[0].from.name');
+        const status = _.get(imageStreamImport, 'status.images[0].status');
+        if (status.status === 'Success') {
+          this.setState({
+            loading: false,
+            isi: {
+              name,
+              image: _.get(imageStreamImport, 'status.images[0].image'),
+              tag: _.get(imageStreamImport, 'status.images[0].tag'),
+              status
+            },
+            name,
+          });
+        } else {
+          this.setState({
+            loading: false,
+            searchError: _.upperFirst(status.message),
+          });
+        }
+      }, err => this.setState({error: err.message}));
+  };
+
+  save = event => {
+    event.preventDefault();
+
+    this.setState({
+      inProgress: true,
+      error: null
+    });
+
+    const { name, namespace, isi } = this.state;
+
+    const annotations = {
+      'openshift.io/generated-by': 'OpenShiftWebConsole'
+    };
+
+    const volumes = [], volumeMounts = [];
+    let volumeNumber = 0;
+    _.forEach(isi.image.dockerImageMetadata.Config.Volumes, (value, path) => {
+      volumeNumber++;
+      const volumeName = `${name}-${volumeNumber}`;
+      volumes.push({
+        name: volumeName,
+        emptyDir: {}
+      });
+      volumeMounts.push({
+        name: volumeName,
+        mountPath: path
+      });
+    });
+
+    const ports = parsePorts(isi.image);
+
+    const promises = [];
+
+    const deploymentConfig = {
```

Comment:
I'd create a helper method for each resource you create that returns a promise. It will keep `save()` from getting too long.